### ISSUE TITLE
Error in load_qm9 tutorial: "Explicit valence for atom # 1 C, 4, is greater than permitted"

### ISSUE THREAD

**Issue by budhilnigam** [deepchem/deepchem#4350]
## 🐛 Bug

**Molecule Sanitization Error:**  
   - When loading the QM9 dataset with `dc.molnet.load_qm9(featurizer='ECFP')`, multiple molecule sanitization errors occur.  
   - The error message includes:  
     ```
     ERROR: Explicit valence for atom # 1 C, 4, is greater than permitted
     ERROR: Could not sanitize molecule ending on line 5703576
     ```
   - This issue seems to be due to valence violations in some molecules.  
   - Also there is a deprecation warning : *DEPRECATION WARNING: please use MorganGenerator*.

## To Reproduce

1. Follow the tutorial **Training_a_Normalizing_Flow_on_QM9_2** which is tutorial 8 under the section **2.Molecular Machine Learning**
2. Go to the code block where we load the dataset of 2500 molecules from the QM9 dataset.

![Image](https://github.com/user-attachments/assets/cbd475d6-d5a3-45f9-ad95-965bf7f39951)

## Expected behavior

The QM9 dataset should load without sanitization errors or handle invalid molecules gracefully.

## Environment

* OS: Google Colab (Ubuntu 22.04 Jammy)
* Python version: 3.11.11
* DeepChem version: 2.8.1.dev
* RDKit version (optional): 2024.09.5
* TensorFlow version (optional): 2.18.0
* PyTorch version (optional):
* Any other relevant information:

**Comment by budhilnigam** [comment]
Also the colab notebook link for the tutorial [Training a Normalizing Flow on QM9](https://deepchem.io/tutorials/training-a-normalizing-flow-on-qm9/) at the website leads to _https://colab.research.google.com/github/deepchem/deepchem/blob/master/examples/tutorials/23_Training_a_Normalizing_Flow_on_QM9.ipynb_ (**Incorrect Link**) while the correct link should be _https://colab.research.google.com/github/deepchem/deepchem/blob/master/examples/tutorials/Training_a_Normalizing_Flow_on_QM9.ipynb_ (**Correct Link**)

**Comment by xy9485** [comment]
I also ran into the same issue for example： ERROR: Explicit valence for atom # 1 C, 4, is greater than permitted
@budhilnigam did you find any solution to this?

**Comment by Sidd-art-h1** [comment]
Hi! I am a Chemical Engineering student and I understand why RDKit is throwing this valence error (sanitization issue). I can fix this by updating the loader to handle the sanitization step correctly. Can I work on this?

**Comment by rbharath** [comment]
This may possibly be relevant https://github.com/deepchem/deepchem/discussions/4706
